movement/kinematics.py

```diff
@@ -324,6 +326,330 @@ def compute_head_direction_vector(
+            elem2: getattr(a, labels_dim).values,
+        }
+    )
+    # Drop any squeezed coordinates
+    return result.squeeze(drop=True)
+
+
+def compute_pairwise_distances(
+    data: xr.DataArray,
+    dim: Literal["individuals", "keypoints"],
+    pairs: dict[str, str | list[str]] | Literal["all"],
+    metric: str | None = "euclidean",
+    **kwargs,
+) -> xr.DataArray | dict[str, xr.DataArray]:
+    """Compute pairwise distances between ``individuals`` or ``keypoints``.
+
+    This function computes the distances between
+    pairs of ``individuals`` (i.e. interindividual distances) or
+    pairs of ``keypoints`` (i.e. interkeypoint distances),
+    as determined by ``dim``.
+    The distances are computed for the given ``pairs``
+    using the specified ``metric``.
+
+    Parameters
+    ----------
+    data : xarray.DataArray
+        The input data containing
+        ``('time', 'individuals', 'keypoints', 'space')`` as dimensions.
+    dim : Literal["individuals", "keypoints"]
+        The dimension to compute the distances for. Must be either
+        ``'individuals'`` or ``'keypoints'``.
+    pairs : dict[str, str | list[str]] | Literal["all"]
+        A dictionary containing the mapping between pairs of ``individuals``
+        or ``keypoints``, or the special keyword ``"all"``.
+
+        - If a dictionary is provided,
+          the key is the keypoint or individual and
+          the value can be a list of keypoints or individuals,
+          or a string representing a single keypoint or individual
+          to compute the distance with.
+        - If the special keyword ``'all'`` is provided,
+          all possible combinations of pairs are computed.
+    metric : str, optional
+        The distance metric to use. Must be one of the options supported
+        by :func:`scipy.spatial.distance.cdist`, e.g. ``'cityblock'``,
+        ``'euclidean'``, etc.
+        Defaults to ``'euclidean'``.
+    **kwargs : dict
+        Additional keyword arguments to pass to
+        :func:`scipy.spatial.distance.cdist`.
+
+    Returns
+    -------
+    xarray.DataArray | dict[str, xarray.DataArray]
+        An :class:`xarray.DataArray` containing the computed distances between
+        a single pair of ``individuals`` or ``keypoints``. If multiple
+        ``pairs`` are provided, this will be a dictionary containing the
+        computed distances for each pair, with the key being the pair of
+        keypoints or individuals and the value being the
+        :class:`xarray.DataArray` containing the computed distances.
+
+    Raises
+    ------
+    ValueError
+        If ``dim`` is not one of ``'individuals'`` or ``'keypoints'``;
+        if ``pairs`` is not a dictionary or ``'all'``; or
+        if there are no pairs in ``data`` to compute distances for.
+
+    Examples
+    --------
+    Compute the Euclidean distance (default) for all keypoints
+    between ``ind1`` and ``ind2`` (i.e. interindividual distance):
```

Comment:
Slightly altered phrasing:

```suggestion
    Compute the Euclidean distance (default) between ``ind1`` and ``ind2``
    (i.e. interindividual distance), for all possible pairs of keypoints.
```